### PR TITLE
Add FreeBSD CI build.

### PR DESCRIPTION
**Pull request by RICCIARDI-Adrien**
It is not as complete as Linux builds but it still does some useful checks.

### CODE REVIEW

**Review by stappersg — COMMENTED** [review]
Why should _FreeBSD_ be build on _MacOS_?

In case the answer is _because that one is available_,  provide the list of options.
(Yes, this comment reveals that I don't know where to find documentation of the _CI stuff_ we are using.)

File: .github/workflows/bsd.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+name: BSD
+on: [push, pull_request]
+
+jobs:
+  freebsd:
+    runs-on: macos-latest
```
[the remark this answers is not in the document]
Comment (RICCIARDI-Adrien) [reply .github/workflows/bsd.yaml:6]: Hi,
I confirm that GitHub Actions documentation is big and not really well organized!

GitHub provides 3 host operating systems : Linux, Windows and macOS (https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources), there is no way to run another host OS for GitHub-hosted runners.

However, virtualization is enabled on macOS machines, so the guy that created the FreeBSD action used in this PR is installing VirtualBox in macOS to run a FreeBSD VM. When the VM is booted, it connects to it though SSH and executes our commands.

I already used a similar action to enable Solaris build in PPP project, it works quite well : https://github.com/ppp-project/ppp/blob/master/.github/workflows/solaris.yaml.

**Comment by stappersg** [comment]
https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources

**Review by stappersg — APPROVED** [review]
OK

**Pull request merged**